spk/rutorrent/src/service-setup.sh

```diff
@@ -0,0 +1,310 @@
+                if [ "`synoacltool -get \"${DIRNAME}\"| grep \"Linux mode\"`" ]; then
+                    synoacltool -add "${DIRNAME}" "group:administrators:allow:rwxpdDaARWc--:fd--" >> ${INST_LOG} 2>&1
+                fi
+                # Add the new group permissions
+                echo "Granting '${KEY}' basic permissions on ${DIRNAME}" >> ${INST_LOG}
+                synoacltool -add "${DIRNAME}" "${KEY}:allow:rxaR:n" >> ${INST_LOG} 2>&1
+            fi
+            DIRNAME="$(dirname "${DIRNAME}")"
+        done
+    else
+        echo "Skip granting '${KEY}' basic permissions on ${DIRNAME} as the directory does not reside in '/volumeX'. Set manually if needed." >> ${INST_LOG}
+    fi
+}
+
+service_preinst ()
+{
+    if [ "${SYNOPKG_PKG_STATUS}" == "INSTALL" ]; then
+        if [ ! -d "${wizard_download_dir}" ]; then
+            echo "Download directory ${wizard_download_dir} does not exist."
+            exit 1
+        fi
+        if [ -n "${wizard_watch_dir}" -a ! -d "${wizard_watch_dir}" ]; then
+            echo "Watch directory ${wizard_watch_dir} does not exist."
+            exit 1
+        fi
+    fi
+
+    exit 0
+}
+
+fix_shared_folders_rights()
+{
+    local folder=$1
+    echo "Fixing shared folder rights for ${folder}" >> "${INST_LOG}"
+    chown -R "${APACHE_USER}:${USER}" "${folder}" >> "${INST_LOG}" 2>&1
+    chmod ug+rwx "${folder}" >> "${INST_LOG}" 2>&1
+    synoacltool -add "${folder}" "user:${EFF_USER}:allow:rwxpdDaARWcC:fd" >> "${INST_LOG}" 2>&1
+    synoacltool -add "${folder}" "user:${APACHE_USER}:allow:rwxpdDaARWcC:fd" >> "${INST_LOG}" 2>&1
+    find "${folder}" -mindepth 1 -type d -exec synoacltool -enforce-inherit "{}" \; >> ${INST_LOG} 2>&1
+}
+
+service_postinst ()
+{
+
+    # Install busybox stuff
+    ${INSTALL_DIR}/bin/busybox --install ${INSTALL_DIR}/bin
+
+    syno_user_add_to_legacy_group "${EFF_USER}" "${LEGACY_USER}" "${LEGACY_GROUP}"
+
+    # Install the web interface
+    cp -pR ${INSTALL_DIR}/share/${PACKAGE} ${WEB_DIR} >>"${INST_LOG}" 2>&1
+
+    # Configure open_basedir
+    if [ "${APACHE_USER}" == "nobody" ]; then
+        echo -e "<Directory \"${WEB_DIR}/${PACKAGE}\">\nphp_admin_value open_basedir none\n</Directory>" > /usr/syno/etc/sites-enabled-user/${PACKAGE}.conf
+    else
+        if [ -d "/etc/php/conf.d/" ]; then
+            echo -e "[PATH=${WEB_DIR}/${PACKAGE}]\nopen_basedir = Null" > /etc/php/conf.d/${PACKAGE_NAME}.ini
+        fi
+    fi
+
+    # Configure files
+    if [ "${SYNOPKG_PKG_STATUS}" == "INSTALL" ]; then
+        TOP_DIR=`echo "${wizard_download_dir:=/volume1/downloads}" | cut -d "/" -f 2`
+        MAX_MEMORY=`awk '/MemTotal/{memory=$2*1024*0.25; if (memory > 512*1024*1024) memory=512*1024*1024; printf "%0.f", memory}' /proc/meminfo`
+
+        sed -i -e "s|scgi_port = 5000;|scgi_port = 8050;|g" \
+               -e "s|topDirectory = '/';|topDirectory = '/${TOP_DIR}/';|g" \
+               -e "s|tempDirectory = null;|tempDirectory = '${INSTALL_DIR}/tmp/';|g" \
```

Comment:
@th0ma7 the issue we faced was due to the value enforced which wasn't ending with a trailing slash... And that's where temporary resources were actually supposed to be stored. Since ruTorrent seems to be using old school concatenation it was giving files directly under the `${INSTALL_DIR}` directory prefixed with `tmp`.